src/tracks/playabletrack/wavetrack/ui/WaveTrackAffordanceControls.cpp

```diff
@@ -20,61 +21,183 @@
+        {
+            mHelper->Cancel(project);
+            mHelper.reset();
+        }
+        return RefreshCode::RefreshAll;
+    }
+};
+
+class WaveTrackAffordanceHandle final : public AffordanceHandle
+{
+    std::shared_ptr<WaveClip> mTarget;
+    bool mDoubleClickFlag{ false };
+public:
+    WaveTrackAffordanceHandle(const std::shared_ptr<Track>& track, const std::shared_ptr<WaveClip>& target) 
+        : AffordanceHandle(track), mTarget(target)
+    { }
+
+    Result Click(const TrackPanelMouseEvent& event, AudacityProject* project) override
+    {
+        mDoubleClickFlag = false;
+        if (event.event.LeftDClick())
+        {
+            //To make sure that Double Click drag/release events are not handled by super class
+            mDoubleClickFlag = true;
+
+            auto& viewInfo = ViewInfo::Get(*project);
+            auto affordanceRect = ClipParameters::GetClipRect(*mTarget.get(), viewInfo, event.rect);
+            if (affordanceRect.Contains(event.event.GetPosition()))
+            {
+                if (auto affordanceControl = std::dynamic_pointer_cast<WaveTrackAffordanceControls>(event.pCell))
+                {
+                    return affordanceControl->StartEditClipName(project) ? 
+                        RefreshCode::RefreshCell : RefreshCode::RefreshNone;
+                }
+            }
+            return RefreshCode::RefreshNone;
         }
+        else
+            return AffordanceHandle::Click(event, project);
+    }
+
+    Result Release(const TrackPanelMouseEvent& event, AudacityProject* project, wxWindow* parent) override
+    {
+        if (mDoubleClickFlag)
+            return RefreshCode::RefreshNone;
+        else
+            return AffordanceHandle::Release(event, project, parent);
+    }
 
-        ProjectHistory::Get(*pProject).ModifyState(false);
+    Result Drag(const TrackPanelMouseEvent& event, AudacityProject* project) override
+    {
+        if (mDoubleClickFlag)
+            return RefreshCode::RefreshNone;
+        else
+            return AffordanceHandle::Drag(event, project);
+    }
 
+    UIHandle::Result SelectAt(const TrackPanelMouseEvent& event, AudacityProject* project) override
+    {
+        auto& viewInfo = ViewInfo::Get(*project);
+        viewInfo.selectedRegion.setTimes(mTarget->GetOffset(), mTarget->GetEndTime());
+        
+        ProjectHistory::Get(*project).ModifyState(false);
+        
         return RefreshCode::RefreshAll | RefreshCode::Cancelled;
     }
 };
 
 WaveTrackAffordanceControls::WaveTrackAffordanceControls(const std::shared_ptr<Track>& pTrack)
-    : CommonTrackCell(pTrack)
+    : CommonTrackCell(pTrack), mClipNameFont(wxFont(wxFontInfo()))
 {
+    if (auto waveTrack = dynamic_cast<WaveTrack*>(pTrack.get()))
+    {
+        //Unbind is unnecessary since wave track is 
+        //destroyed along with affordance cell
+        waveTrack->Bind(EVT_WAVETRACK_SELECTION,
+            &WaveTrackAffordanceControls::OnTrackChanged,
+            this);
```

Comment:
Unbind() is also unnecessary when the event "sink" object in the Bind() call also inherits wxEvtHandler.  Then when it is destroyed, the "source" wxEvtHandler is also notified.

(I'm not certain, but it may in fact be enough for the sink to inhert wxTrackable, a base class of wxEvtHandler.)

Can the text edit helper inherit wxEvtHandler and bind itself to the TrackList?  Then it can destroy itself in response to the event.